### PR TITLE
WIP: DD-DeCaF pathway predictor

### PR DESCRIPTION
**Pull request by sgalkina**

### CODE REVIEW

File: resolwe_bio/processes/modeling/pathway_predictor.yml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
         image: dddecaf/resolwe-pathway-predictor:latest
+    resources:
+      network: true
+      cores: 4
   data_name: "Predict heterologous metabolic pathways"
   version: 0.0.1
-  type: data:similarexpression
+  type: data:pathways
   persistence: TEMP
   scheduling_class: interactive
```
Comment (jkokosar) [inline resolwe_bio/processes/modeling/pathway_predictor.yml]: Processes with ``scheduling_class: interactive`` are run in a separate queue and are killed after 30s if not finished in time. This happens in the test case already, so you probably need to remove this.

File: resolwe_bio/tests/processes/test_pathway_predictor.py [hunk 2]
```diff
@@ -4,9 +4,9 @@
...
-    @tag_process('pathway-predictor')
+class PathwayPredictorTestCase(ProcessTestCase):
+    @tag_process('pathways-predictor')
     def test_pathway_predictor(self):
-        inputs = {'model': 'one'}
-        annotation = self.run_process('pathway-predictor', inputs)
-        self.assertFields(annotation, 'pathway', 'l')
+        inputs = {'model': 'iMM904', 'product': 'methanol', 'n_pathways': 2}
+        annotation = self.run_process('pathways-predictor', inputs)
+        self.assertFields(annotation, 'pathway', '')
```
Comment (jkokosar) [inline resolwe_bio/tests/processes/test_pathway_predictor.py]: (annotation, 'pathway', '') -> (annotation, 'pathway**s**', '**test_output.json**')

File: resolwe_bio/tools/pathway_predictor.py [hunk 3]
```diff
@@ -1,7 +1,26 @@
 #!/usr/bin/env python3
```
Comment (jkokosar) [inline resolwe_bio/tools/pathway_predictor.py]: It much better to save all the output to a json file and than call the ``re-save pathways my_file.json`` in the process itself. For each addition to RESULTS list, you could print out something like ``print('{"proc.progress":0.3}'``. The progress value will show in the bioinfo dashboard as an analysis progress (30 % done in that case).

File: resolwe_bio/tools/pathway_predictor.py [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+    return dict(
+        id=reaction.id,
+        name=reaction.name,
+        reaction_string=reaction.build_reaction_string(use_metabolite_names=True),
+    )
+
+
+def show_results(pathway):
+    RESULTS.append([reaction_to_dict(reaction) for reaction in pathway.reactions])
+    call(['re-save', 'pathways', json.dumps(RESULTS)])
```
Comment (jkokosar) [inline resolwe_bio/tools/pathway_predictor.py]: [This ](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/tools/geneset2storage.py#L30) is an example of code that would create an output json file to be saved with `re-save` in the .yml file

File: resolwe_bio/tools/pathway_predictor.py [hunk 5]
```diff
@@ -1,7 +1,26 @@
...
+
+def reaction_to_dict(reaction):
+    return dict(
+        id=reaction.id,
+        name=reaction.name,
+        reaction_string=reaction.build_reaction_string(use_metabolite_names=True),
+    )
+
+
+def show_results(pathway):
```
Comment (jkokosar) [inline resolwe_bio/tools/pathway_predictor.py]: Where do you use the function?

File: tests/settings.py [hunk 6]
```diff
@@ -125,6 +125,9 @@
     'resolwe.flow.execution_engines.workflow',
 ]
 
+NOSE_ARGS = ['--nocapture',
```
Comment (jkokosar) [inline tests/settings.py]: Why is this needed?

File: resolwe_bio/processes/modeling/pathway_predictor.yml [hunk 7]
```diff
@@ -1,13 +1,16 @@
-- slug: pathway-predictor
+- slug: pathways-predictor
   name: Predict heterologous metabolic pathways
   requirements:
     expression-engine: jinja
     executor:
       docker:
         image: dddecaf/resolwe-pathway-predictor:latest
```
Comment (jkokosar) [inline resolwe_bio/processes/modeling/pathway_predictor.yml]: `dddecaf/resolwe-pathway-predictor`

latest is the default, thus not needed here

File: resolwe_bio/processes/modeling/pathway_predictor.yml [hunk 8]
```diff
@@ -1,13 +1,16 @@
...
+    resources:
+      network: true
+      cores: 4
   data_name: "Predict heterologous metabolic pathways"
```
Comment (jkokosar) [inline resolwe_bio/processes/modeling/pathway_predictor.yml]: You can construct a more informative data_name out of process inputs. [Example](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/go_enrichment/go.yml#L15) that includes gene names in the data_name

**Comment by jkokosar** [comment]
Also, better organize the commits. Squash together what is necessary (modifications to the same file/lines). Add separate commit for settings change. Docker file addition should be in its own pull request (this is a limitation of the current Jenkins testing framework implementation).

Add changelog entry.

File: resolwe_bio/tools/pathway_predictor.py (hunk 3 above)
Comment (sgalkina) [reply resolwe_bio/tools/pathway_predictor.py]: but in this case the json file would be saved to MongoDB only after the script finished running, and what is needed is that the entry in MongoDB is updated with new set of reactions during the script execution

**Comment by mstajdohar** [comment]
We have fixed the testing issues on resolwe-bio, so we can now merge your pull request. @jkokosar Please do another review pass. @sgalkina After Janes'z review, please rebase to master.
